### DISCUSSION TITLE
[Feature] Support multiple machine learning service endpoints

### DISCUSSION THREAD

**Discussion by Harry-Chen** [immich-app/immich#8589] [Feature Request]
### The feature

ML tasks are compute-intensive, so users may want to enhance parallelism by adding more devices. It is easy to spawn multiple ML services since they are independent of each other, but the immich server itself could only use one at the moment.

I suggest that the main server add support for multiple ML service endpoints, and use them in a load-balanced way (simple round-robin or user-specified priority, this could be discussed later). I am aware that one ML service could accept multiple connections and handle them in a thread pool. But there are always situations in which users have separate devices / machines to utilize.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Duplicate of #4616
